cli/src/main/java/com/devonfw/tools/ide/environment/AbstractEnvironmentVariables.java

```diff
@@ -132,7 +133,7 @@ private final Collection<VariableLine> collectVariables(boolean onlyExported) {
   /**
    * @param variables the {@link Set} where to add the names of the variables defined here.
    */
-  protected void collectVariables(Set<String> variables) {
+  protected void collectVariables(Map<String, String> variables) {
```

Comment:
Why was this changed from a Set (prevents duplicates) to a Map?

Reply:
Since the variables are the keys of the map, duplicates are still prevented. A map now allows us to additionally keep track of the source of the variable, which is contained in the value part of the map.

Reply:
Ah, ok. I was too focused on HashMaps allowing duplicate values but missed the key. At least the Javadoc got corrected now, missed this too...